framework/wazuh/core/wazuh_queue.py

```diff
@@ -164,5 +182,49 @@ def send_msg_to_agent(self, msg: str = '', agent_id: str = '', msg_type: str = '
 
         return ret_msg
 
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        self.close()
+
+class WazuhAnalysisdQueue(BaseQueue):
+    """
+    WazuhAnalysisdQueue Object.
+    """
+
+    MAX_MSG_SIZE = 65535
+
+    def _connect(self):
+        try:
+            self.socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+            self.socket.connect(self.path)
+        except Exception:
+            raise WazuhInternalError(1010, self.path)
+
+    def send_msg(self, msg_header: str, msg: str):
+        """Send message to analysisd.
+
+        Parameters
+        ----------
+        msg_header : str
+            Header message to attach.
+        msg : str
+            Message to send.
+
+        Raises
+        ------
+        WazuhInternalError(1012)
+            If the size of the event is bigger than the message size that analysisd can handle.
+        WazuhError(1014)
+            If there was an error communicating with socket.
+        """
+        socket_msg = f"{msg_header}{msg}".encode()
+
+        if len(socket_msg) > self.MAX_MSG_SIZE:
+            raise WazuhError(
+                1012,
+                "The size of the event is bigger than the message size that analysisd can handle"
+                f"({self.MAX_MSG_SIZE}b)."
```

Comment:
Max size is bytes (`B`), not bits (`b`). I think these message is a bit easier to understand:
```suggestion
                "The event is too large to be sent to analysisd (maximum is {self.MAX_MSG_SIZE}B)"
```